### PR TITLE
Hotspot fix for char pointer and sonarscan bugs report fix

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
Hotspot fix for char pointer
Add return for operator override function.

## Related Issue
NA

## Motivation and Context
UC#1

## How Has This Been Tested?
Local integration testing

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by paulbourelly999** [comment]
@dan-du-car could you rebase your branch on develop so you get the sonar-scanner changes. This way we can get accurate unit test coverage and static code analysis.

File: message_services/src/models/bsm.cpp [hunk 1]
```diff
@@ -209,11 +209,12 @@ namespace message_services
 
         std::ostream &operator<<(std::ostream &out, bsm &bsm_obj)
         {
-            std::cout << std::fixed;
-            std::cout << std::setprecision(6);
-            std::cout << "bsm model: core_data: { temprary_id = " << bsm_obj.core_data.temprary_id << ", sec_mark = " << bsm_obj.core_data.sec_mark
+            out << std::fixed;
+            out << std::setprecision(6);
+            out << "bsm model: core_data: { temprary_id = " << bsm_obj.core_data.temprary_id << ", sec_mark = " << bsm_obj.core_data.sec_mark
```
Comment (paulbourelly999) [inline message_services/src/models/bsm.cpp:214]: What is the output stream for? Should we be logging any of this with spd logger?

File: message_services/src/models/vehicle_status_intent.cpp [hunk 2]
```diff
@@ -159,7 +159,8 @@ namespace message_services
 
         std::ostream &operator<<(std::ostream &out, vehicle_status_intent &vehicle_status_intent_obj)
         {
-            std::cout << "vehicle_status_intent model: vehicle_id  = " << vehicle_status_intent_obj.vehicle_id << ", current_timestamp = " << vehicle_status_intent_obj.cur_timestamp << ", bsm_id =" << vehicle_status_intent_obj.bsm_id << ", current speed= " << vehicle_status_intent_obj.cur_speed << std::endl;
+            out << "vehicle_status_intent model: vehicle_id  = " << vehicle_status_intent_obj.vehicle_id << ", current_timestamp = " << vehicle_status_intent_obj.cur_timestamp << ", bsm_id =" << vehicle_status_intent_obj.bsm_id << ", current speed= " << vehicle_status_intent_obj.cur_speed << std::endl;
```
Comment (paulbourelly999) [inline message_services/src/models/vehicle_status_intent.cpp:162]: See comment above

File: message_services/src/models/bsm.cpp (hunk 1 above)
Comment (dan-du-car) [reply message_services/src/models/bsm.cpp:214]: It is not necessary to log it. But if any information within the model is useful or required, then we might log it.

**Comment by dan-du-car** [comment]
> @dan-du-car could you rebase your branch on develop so you get the sonar-scanner changes. This way we can get accurate unit test coverage and static code analysis.

@paulbourelly999 As we discussed, please merge this first. If the issues this PR shall address is not cleared after merge, I will create another PR to address it.

**Review by paulbourelly999 — APPROVED** [review]

**Pull request merged**